src/PixiEditor/Models/DocumentModels/DocumentUpdater.cs

```diff
@@ -511,28 +505,43 @@ private void ClearSelectedKeyFrames(ClearSelectedKeyFrames_PassthroughAction inf
         doc.AnimationHandler.ClearSelectedKeyFrames();
     }
 
-    private void ProcessCreateNode<T>(CreateNode_ChangeInfo info) where T : NodeViewModel, new()
+    private void ProcessCreateNode(CreateNode_ChangeInfo info)
     {
-        T node = new T()
-        {
-            InternalName = info.InternalName, Id = info.Id, Document = (DocumentViewModel)doc, Internals = helper
-        };
+        var nodeType = info.Metadata.NodeType;
+        
+        var ns = nodeType.Namespace.Replace("PixiEditor.ChangeableDocument.Changeables.Graph.Nodes", "PixiEditor.ViewModels.Document.Nodes");
```

Comment:
I hate this being hardcoded, maybe query last dot and substring the type name, then append "PixiEditor.Viewmodels.Document.Nodes" there?